### PR TITLE
ebpf memory cleanup

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
This PR fixes the memory cleanup when `ebpf.plugin` is closed.

I am also using this PR to remove some unnecessary code, because we are using `libbpf` now, but instead to remove the old code, I decided to comment it, because it can be useful where `libbpf` cannot be used.

##### Component Name
Ebpf plugin
##### Test Plan

1 - Install `libasan` on your environment. 
2 - Compile this branch with the following flags: 

```
CFLAGS="-O1 -ggdb -Wall -Wextra -fsanitize=address -static-libasan -fno-omit-frame-pointer -Wformat-signedness -fstack-protector-all -DNETDATA_INTERNAL_CHECKS=1 -D_FORTIFY_SOURCE=2 -DNETDATA_VERIFY_LOCKS=1" ./netdata-installer.sh --disable-lto --dont-wait --dont-start-it
```

If you prefer you can use the dynamic `libasan`.

3 - Go to directory `/usr/libexec/netdata/plugins.d`
4 - Run `ebpf.plugin` redirecting `STDERR` and `STDOUT` for files:

```
./ebpf.pluging 2> stderr > stdout
```

5 - After few minutes, kill the process and investigate your `stderr` file, you cannot have any memory leak reported.
6 - Start Netdata and take a look on `ebpf.plugin` charts.
7 - Access `apps.plugin` and also take a look on the ebpf section to confirm that charts are ok.

##### Additional Information

I tested this PR with the the kernels `5.9.1`, `5.8.12`, `5.4.72`, `4.19.144`, `4.17.19`, `4.15.0`, `4.14.197`. I observed that the network monitoring is not working fine for kernels older than `4.17`, but this is a known problem that will be fixed next days, for the newest kernels everything is working as expected.

### CODE REVIEW

File: collectors/ebpf.plugin/ebpf_process.c [hunk 1]
```diff
@@ -866,24 +872,77 @@ static void process_collector(usec_t step, ebpf_module_t *em)
  *
  *****************************************************************/
 
+void clean_global_memory() {
+    int pid_fd = map_fd[0];
+    struct pid_stat *pids = root_of_pids;
+    while (pids) {
+        uint32_t pid = pids->pid;
+        ebpf_process_stat_t *w = global_process_stats[pid];
+        freez(w);
```
Comment (vlvkobal) [inline collectors/ebpf.plugin/ebpf_process.c:881]: You don't need `w` here. Use `global_process_stats[pid]` directly.

File: libnetdata/ebpf/ebpf.c [hunk 2]
```diff
@@ -280,40 +282,41 @@ static int select_file(char *name, const char *program, size_t length, int mode,
     return ret;
 }
 
-int ebpf_load_program(char *plugins_dir, int event_id, int mode, char *kernel_string, const char *name, int *map_fd)
+//give ojs as arg and get link as result
```
Comment (vlvkobal) [inline libnetdata/ebpf/ebpf.c:285]: A typo: `ojs`.

File: libnetdata/ebpf/ebpf.c [hunk 3]
```diff
@@ -280,40 +282,41 @@ static int select_file(char *name, const char *program, size_t length, int mode,
...
 
     snprintf(lpath, 4096, "%s/%s", plugins_dir, lname);
-    if (bpf_prog_load(lpath, BPF_PROG_TYPE_KPROBE, &obj, &prog_fd)) {
+    if (bpf_prog_load(lpath, BPF_PROG_TYPE_KPROBE, obj, &prog_fd)) {
         info("Cannot load program: %s", lpath);
-        return -1;
+        return NULL;
     } else {
-        info("The eBPF program %s was loaded with success.", name);
+        info("The eBPF program %s was loaded with success.",em->thread_name);
```
Comment (vlvkobal) [inline libnetdata/ebpf/ebpf.c:301]: Missing whitespace.

**Comment by thiagoftsm** [comment]
@vlvkobal I made all changed you requested.
I also tested with `valgrind` instead `libasan` and I can confirm we are having the same results.

**Review by vlvkobal — APPROVED** [review]

**Review by underhood — APPROVED** [review]

**Pull request merged**